src/messaging/ExchangeContext.h

```diff
@@ -175,6 +175,15 @@ class DLL_EXPORT ExchangeContext : public ReliableMessageContext,
     void Close();
     void Abort();
 
+    // Applies a suggested response timeout value based on the session type and the given upper layer processing time for
+    // the next message to the exchange. The exchange context must have a valid session when calling this function.
+    //
+    // This function is an equivalent of SetResponseTimeout(mSession->SuggestEndToEndTimeout(applicationProcessingTimeout))
+    void UseSuggestedResponseTimeout(Timeout applicationProcessingTimeout);
```

Comment:
I'm not certain having this convenience is a good thing - I think callers should explicitly call into the session to get the computed timeout and just call `SetResponseTimeout` themselves. We want to not hide away these details - callers should explicitly ask the session object to get an adjusted timeout if they want to.

Reply:
sessions are hidden behind the exchanges -- the client will create an exchange context from a session, and use the exchange context in the remaining time, and for servers, it will receive an exchange instead of a session.

Since in most cases, the caller will not and is not expected to use an "raw" (unadjusted) timeout, so the most common case for setting the timeout will be, `SetResponseTimeout(mSession->ComputeRoundTripTimeout(applicationProcessingTimeout))`.

Reply:
Ok, I could be convinced to leave it in for now!